### PR TITLE
ES Lint

### PR DESCRIPTION
**Pull request by GentBinakuSQA**

### CODE REVIEW

**Comment by lewis-sqa** [comment]
I know this is still in draft, but we should look to create a separate PR to address the formatting issues that ESLint has detected. It makes it easy to review that way. As mentioned in the task, we'll want to time the formatting changes with the team to avoid triggering excessive Git conflicts.

**Comment by GentBinakuSQA** [comment]
@lewis-sqa 
https://github.com/near-projects/wallet-selector/blob/7e51d458d80d4572f6648cd9da9a735b621a7511/package.json#L12

File: package.json [hunk 1]
```diff
@@ -9,9 +9,15 @@
...
     "build:cjs": "tsc --module commonjs --outDir lib/cjs",
+    "lint": "eslint \"./**/*.ts\"  \"./**/*.tsx\"",
+    "lint:fix": "eslint \"./**/*.ts\"  \"./**/*.tsx\" --fix",
     "test": "jest",
     "prepare": "npm run build"
   },
+  "eslintConfig": {
+    "root": true,
+    "extends": "@ntnyq/prettier"
+  },
```
Comment (lewis-sqa) [inline package.json:20]: We don't need this configuration as ESLint will use the dedicated file

File: package.json [hunk 2]
```diff
@@ -37,7 +43,15 @@
...
     "@types/w3c-web-hid": "^1.0.2",
     "@types/w3c-web-usb": "^1.0.5",
+    "@typescript-eslint/eslint-plugin": "^5.10.1",
+    "@typescript-eslint/parser": "^5.10.1",
+    "eslint": "^8.7.0",
+    "eslint-config-prettier": "^8.3.0",
+    "eslint-plugin-prettier": "^4.0.0",
+    "install": "^0.13.0",
     "jest": "^27.4.7",
+    "npm": "^8.4.0",
```
Comment (lewis-sqa) [inline package.json:53]: We can remove this dependency as this supplied with Node.js itself and considered a prerequisite for this project.

File: README.md [hunk 3]
```diff
@@ -67,25 +67,32 @@ Add event listeners (init, disconnect, signIn):
...
+    console.log(messages);
+  });
 ```
 
 ## Example Integration
 
-A variation of the [guest-book](https://github.com/near-examples/guest-book/)  example project can be found in the `example` directory. You can use this to gain a concrete understanding of how to integrate this package into your own dApp.
+A variation of the [guest-book](https://github.com/near-examples/guest-book/) example project can be found in the `example` directory. You can use this to gain a concrete understanding of how to integrate this package into your own dApp.
 
+<<<<<<< HEAD
```
Comment (lewis-sqa) [inline README.md:89]: It looks like some conflicts are still present here. As mentioned previously however, changes from `npm run lint:fix` should be done as part of another PR. The goal of this PR is just to enable ESLint without impacting the entire codebase

**Review by lewis-sqa — CHANGES_REQUESTED** [review]

**Comment by lewis-sqa** [comment]
Superseded by #105.

**Pull request closed without merging**